migrate.go

```diff
@@ -389,34 +395,71 @@ func PlanMigration(db *sql.DB, dialect string, m MigrationSource, dir MigrationD
 	return result, dbMap, nil
 }
 
-// Filter a slice of migrations into ones that should be applied.
-func ToApply(migrations []*Migration, current string, direction MigrationDirection) []*Migration {
-	var index = -1
-	if current != "" {
-		for index < len(migrations)-1 {
-			index++
-			if migrations[index].Id == current {
+// Searches the arrays for the latest elements with common ids.
+// Returns the indexes of the common id.
+func lastCommon(x []*Migration, y []*Migration) (int, int) {
```

Comment:
Rename `x` and `y` to `left` and `right` to avoid unreadable array accesses like `x[i]` and `y[j]`.